src/main/java/org/opensearch/sdk/SDKTransportService.java

```diff
@@ -0,0 +1,126 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.Nullable;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.extensions.action.RegisterTransportActionsRequest;
+import org.opensearch.extensions.action.TransportActionRequestFromExtension;
+import org.opensearch.sdk.ActionExtension.ActionHandler;
+import org.opensearch.sdk.action.ProxyActionRequest;
+import org.opensearch.sdk.action.SDKActionModule;
+import org.opensearch.sdk.handlers.AcknowledgedResponseHandler;
+import org.opensearch.sdk.handlers.ExtensionActionResponseHandler;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Wrapper class for {@link TranspoortService} and associated methods.
+ *
+ * TODO: Move all the sendFooRequest() methods here
+ * TODO: Replace usages of getExtensionTransportService with this class
+ * https://github.com/opensearch-project/opensearch-sdk-java/issues/585
+ */
+public class SDKTransportService {
+    private final Logger logger = LogManager.getLogger(SDKTransportService.class);
+
+    private TransportService transportService;
+    private DiscoveryNode opensearchNode;
+    private String uniqueId;
+
+    /**
+     * Requests that OpenSearch register the Transport Actions for this extension.
+     *
+     * @param actions The map of registered actions from {@link SDKActionModule#getActions()}
+     */
+    public void sendRegisterTransportActionsRequest(Map<String, ActionHandler<?, ?>> actions) {
+        logger.info("Sending Register Transport Actions request to OpenSearch");
+        Set<String> actionNameSet = actions.values().stream().map(h -> h.getAction().getClass().getName()).collect(Collectors.toSet());
+        AcknowledgedResponseHandler registerTransportActionsResponseHandler = new AcknowledgedResponseHandler();
+        try {
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsManager.REQUEST_EXTENSION_REGISTER_TRANSPORT_ACTIONS,
+                new RegisterTransportActionsRequest(uniqueId, actionNameSet),
+                registerTransportActionsResponseHandler
+            );
+        } catch (Exception e) {
+            logger.info("Failed to send Register Transport Actions request to OpenSearch", e);
+        }
+    }
+
+    /**
+     * Requests that OpenSearch execute a Transport Actions on another extension.
+     *
+     * @param request The request to send
+     * @return A buffer serializing the response from the remote action if successful, otherwise null
+     */
+    @Nullable
```

Comment:
Great question, @vibrantvarun .  I'm working up a sequence diagram that should explain it, but for now here's a text-based version:
 - Anywhere in code that you want to call a remote action you will create a `ProxyActionRequest`.  In this PR you can start in the `RestRemoteHelloAction` but it doesn't need to be a REST call
 - The `ProxyActionRequest` contains 3 bits of information. If you have compile-time access to the classes, constructing it is simplified:
   - The remote action class name
   - The remote request class name
   - The remote request parameters as bytes to be used with the request's StreamInput constructor
 - `client.execute()` is called with the ProxyAction instance and the request generated in the previous step.  This causes the code in `ProxyTransportAction` to be invoked on the request.
 - `ProxyTransportAction.doExecute()` calls `sdkTransportService.sendProxyActionRequest(request)` (this method).
 - Sending the request (this method) copies info into a `TransportActionRequestFromExtension` which OpenSearch expects.  It will know from the action class name which extension has registered it and forwards it there.  It also glues together the request class name and request class bytes into a single byte array that is forwarded to the remote extension (in our hello world case it comes right back!)
 - On the remote extension, the `ExtensionActionRequestHandler` handles the inbound request.  See the `handleExtensionActionRequest()` method which has several comments showing how it finds the action to execute, reconstructs the appropriate request object from the bytes, runs it, and turns its response into bytes that get relayed back to OpenSearch and back to the originating extension.
 - Throughout this process there is exception handling
 - Back at the original sending point, if the bytes are received, they need to be used to construct the corresponding `Response` object for the original request and action.  This is done in the `RestRemoteHelloAction`.

Reply:
@vibrantvarun see https://github.com/opensearch-project/opensearch-sdk-java/pull/588/files#diff-7e38c8f69d56fdabc2d8abee876b213b9a230a1b42a24239bf42c019a19e7aba